flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -114,6 +118,9 @@
 
 	private final Map<ResourceProfile, Integer> resourcePriorities = new HashMap<>();
 
+	/** The containers that we expected to register with ResourceManager. */
+	private final Map<ResourceID, Container> pendingContainersExpectedToRegister = new ConcurrentHashMap<>();
```

Comment:
A concurrent map is not needed because state modifications are performed in the main thread of the RPC framework.

It's enough to use a `HashMap`:
```
private final Map<ResourceID, Container> pendingContainersExpectedToRegister = new HashMap<>();
```

Reply:
I not sure whether a `HashMap` is enough here, because the `pendingContainersExpectedToRegister` not only used in the API that driven by RPC, but also used in some API that related to yarn itself, like the `onContainersCompleted()`, `onContainersAllocated()`. I made it to concurrent map also because I notified that `workerNodeMap` is also a concurrent map which has a similar behavior with `pendingContainersExpectedToRegister`, what do you think?